pkg/destroy/powervs/cloudconnection.go

```diff
@@ -1,24 +1,44 @@
 package powervs
 
 import (
+	"context"
 	"github.com/IBM-Cloud/power-go-client/power/models"
 	"github.com/pkg/errors"
 	"log"
 	"strings"
+	"time"
 )
 
 // listCloudConnections lists cloud connections in the cloud.
 func (o *ClusterUninstaller) listCloudConnections() (cloudResources, error) {
-	// https://github.com/IBM-Cloud/power-go-client/blob/v1.0.88/power/models/cloud_connections.go#L20-L25
-	var cloudConnections *models.CloudConnections
+	var (
+		ctx context.Context
 
-	// https://github.com/IBM-Cloud/power-go-client/blob/v1.0.88/power/models/cloud_connection.go#L20-L71
-	var cloudConnection *models.CloudConnection
+		// https://github.com/IBM-Cloud/power-go-client/blob/v1.0.88/power/models/cloud_connections.go#L20-L25
+		cloudConnections *models.CloudConnections
 
-	// https://github.com/IBM-Cloud/power-go-client/blob/v1.0.88/power/models/job_reference.go#L18-L27
-	var jobReference *models.JobReference
+		// https://github.com/IBM-Cloud/power-go-client/blob/v1.0.88/power/models/cloud_connection.go#L20-L71
+		cloudConnection *models.CloudConnection
 
-	var err error
+		// https://github.com/IBM-Cloud/power-go-client/blob/v1.0.88/power/models/job_reference.go#L18-L27
+		jobReference *models.JobReference
+
+		err error
+
+		cloudConnectionID string
+
+		// https://github.com/IBM-Cloud/power-go-client/blob/v1.0.88/power/models/cloud_connection_endpoint_v_p_c.go#L19-L26
+		EndpointVpc *models.CloudConnectionEndpointVPC
+
+		// https://github.com/IBM-Cloud/power-go-client/blob/v1.0.88/power/models/cloud_connection_v_p_c.go#L18-L26
+		Vpc *models.CloudConnectionVPC
+
+		foundOne       bool = false
+		foundVpc       bool = false
+		vpcStillExists bool = true
+	)
+
+	ctx, _ = o.contextWithTimeout()
```

Comment:
The docs for `context` say
```
Failing to call the CancelFunc leaks the child and its children until the parent is canceled or the timer fires.
```

Reply:
```
hamzy:hamzy-installer$ grep -r o.contextWithTimeout ./pkg/destroy/powervs/ | wc -l
38
```
Since this is all over the place, how about if I address this in a separate PR?

Reply:
Why not both? Since you're already adding these lines, add them with the `defer cancel()` and send a new PR fixing the other places.